disk-watchdog.c

```diff
@@ -0,0 +1,273 @@
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+#include <string.h>
+#include <getopt.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <systemd/sd-daemon.h>
+
+#define DEFAULT_BLOCK_SIZE 512
+#define DEFAULT_INTERVAL 10000  /* 10ms in microseconds */
+
+static volatile int running = 1;
+static char *test_file = NULL;
+static int interval_us = DEFAULT_INTERVAL;
+static int verbose_mode = 0;
+static int debug_mode = 0;
+static int block_size = DEFAULT_BLOCK_SIZE;
```

Comment:
@alexgg , now you can add the block size. It's called like so in the systemd job:

`ExecStart=/bin/sh -c '@DAEMON_PATH@ -v -f @WD_TEST_FILE@ -b $(blockdev --getss /dev/disk/by-state/active)'
`